### PR TITLE
Resolve: Toggle dark/light mode

### LINKED ISSUE #77: Toggle dark/light mode
**Guidelines**
- Tailwind CSS - Core Concepts - Dark Mode https://tailwindcss.com/docs/dark-mode
- Theme toggle using Tailwind CSS in just 3 steps https://shreyvijayvargiya26.medium.com/theme-toggle-using-tailwind-css-in-just-3-steps-d92e1af6630d
- Learning Leptos: v0.1.0, cargo-leptos, & the World's Best Dark-Mode Toggle https://www.youtube.com/watch?v=AD3FHodVgE8

### PR DESCRIPTION
**Pull request by sectore**
[Screencast from 16.04.2023 15:45:21.webm](https://user-images.githubusercontent.com/47693/232315920-1b9f3e80-a234-4359-81f6-209409a1fbba.webm)

- [x] Add `ThemeToggle` button to header
- [x] Provide `ThemeContext` to handle state
- [x] Store theme persistant in `cookie` - needed because of `SSR` and to avoid flickering

This approach is heavily inspired by [Learning Leptos: v0.1.0, cargo-leptos, & the World's Best Dark-Mode Toggle](https://www.youtube.com/watch?v=AD3FHodVgE8 ) and https://github.com/benwis/benwis_leptos/ 

Closes #77

### CODE REVIEW

**Comment by FelixWeis** [comment]
looks good at first glance! @larrysalibra or @LightRider5 can you review?

**Review by fjahr — COMMENTED** [review]
I can't really run the the environment in the same way I used to before, not sure if this is related to this change or not...
```
$ just watch
just watch-leptos
cargo leptos watch -v --hot-reload
error: unexpected argument '--hot-reload' found
```

File: src/app.rs [hunk 1]
```diff
@@ -64,9 +66,29 @@ pub async fn clear_server_count() -> Result<i32, ServerFnError> {
     Ok(0)
 }
 
+#[server(SetDarkTheme, "/api")]
+pub async fn set_dark_theme(cx: Scope, is_dark: bool) -> Result<bool, ServerFnError> {
```
[the remark this answers is not in the document]
Comment (sectore) [reply src/app.rs:70]: Same warning here. Not sure why, because `cx` is still needed in line 75.

**Comment by fjahr** [comment]
The body size issue is fixed now but the dark background doesn't appear anymore in the main part of the view. Text color changes for example but bg stays white.

Screenshot 2023-04-17 at 3 52 17 PM

**Comment by fjahr** [comment]
ACK, works as expected now

**Pull request merged**